### PR TITLE
fix: dynamically include scale in bounding client rect

### PR DESCRIPTION
**Pull request by tholmes-rms**
This only factors scale into the bounding client rect if the offset parent is also scaled.

Honestly, this doesn't feel particularly elegant and I don't know this stuff intimately enough to foresee any potential side effects, but the existing tests pass... 🤷

### CODE REVIEW

File: src/dom-utils/getCompositeRect.js [hunk 1]
```diff
@@ -8,16 +8,26 @@ import getWindowScrollBarX from './getWindowScrollBarX';
 import getDocumentElement from './getDocumentElement';
 import isScrollParent from './isScrollParent';
 
+function isElementScaled(element) {
+  const rect = element.getBoundingClientRect();
+  const scaleX = rect.width / element.offsetWidth || 1;
+  const scaleY = rect.height / element.offsetHeight || 1;
+
+  return scaleX !== 1 || scaleY !== 1;
+}
```
Comment (tholmes-rms) [inline src/dom-utils/getCompositeRect.js:17]: Not sure if we'd want this pulled out into a util file?

File: src/dom-utils/getCompositeRect.js [hunk 2]
```diff
@@ -8,16 +8,26 @@ import getWindowScrollBarX from './getWindowScrollBarX';
 import getDocumentElement from './getDocumentElement';
 import isScrollParent from './isScrollParent';
 
+function isElementScaled(element) {
+  const rect = element.getBoundingClientRect();
```
Comment (atomiks) [inline src/dom-utils/getCompositeRect.js:12]: There should be a util for this method

Comment (tholmes-rms) [reply src/dom-utils/getCompositeRect.js:12]: The util just adds `x` and `y` to the `rect` returned from `element.getBoundingClientRect` (which we don't need) and optionally does the scaling stuff (which we definitely don't want here). 

So, we could but, we don't _need_ to.

**Comment by tholmes-rms** [comment]
@FezVrasta any thoughts on this?

**Comment by FezVrasta** [comment]
Hey sorry, I'm having some busy weeks. I'll try to take a look as soon as possible

**Comment by tholmes-rms** [comment]
Ok, cool. I'm just pinging :trollface:

**Review by bufordeeds — APPROVED** [review]

**Comment by bufordeeds** [comment]
Want just poking around. Please disregard.

**Comment by tholmes-rms** [comment]
bump

**Comment by FezVrasta** [comment]
Umh looks like Flow exploded upon merge :-(

**Comment by tholmes-rms** [comment]
😬

**Pull request merged**